05.wc/wc.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def parse_options
+  options = { lines: false, words: false, bytes: false }
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: wc.rb [options] [file ...]'
+    opts.on('-l', 'Count lines') { options[:lines] = true }
+    opts.on('-w', 'Count words') { options[:words] = true }
+    opts.on('-c', 'Count bytes') { options[:bytes] = true }
+  end.parse!
+  options.values.any? ? options : options.transform_values { true }
+end
+
+def count_lines(content)
+  content.count("\n")
+end
+
+def count_words(content)
+  content.split(/\s+/).reject(&:empty?).count
+end
+
+def count_bytes(content)
+  content.bytesize
+end
+
+def output_results(counts, options, filename = nil)
+  output = []
+  output << format('%8d', counts[:lines]) if options[:lines]
+  output << format('%8d', counts[:words]) if options[:words]
+  output << format('%8d', counts[:bytes]) if options[:bytes]
+  output << filename unless filename.nil?
+  puts output.join(' ')
+end
+
+def handle_multiple_files(files, options)
+  total_counts = files.map { |file| read_and_count_file_contents(file, options) }
+  return unless files.size > 1
+
+  total_lines, total_words, total_bytes = total_counts.map do |counts|
+    [counts[:lines], counts[:words], counts[:bytes]]
+  end.transpose.map(&:sum)
+
+  total_counts_summary = { lines: total_lines, words: total_words, bytes: total_bytes }
```

Comment:
ちょっと技巧的なのでシンプルに

```suggestion
  total_counts = {
    lines: # lineの合計値を算出,
    words: # ...,
    bytes: # ...
  }
```

みたいな実装がいいかなと思います。

あとtotalにはsummaryのニュアンスが含まれるので、summaryはなくしていいんじゃないでしょうか
（というか、38行目のtotal_countsの命名がおかしいかも？）

Reply:
@JunichiIto 
このように修正しました。total_countsについてはそれより良い変数名が思いつかなかったのでそのままにしています。
```ruby
def calculate_totals(file_counts)
  total_counts = {
    lines: file_counts.sum { |counts, _| counts[:lines] },
    words: file_counts.sum { |counts, _| counts[:words] },
    bytes: file_counts.sum { |counts, _| counts[:bytes] }
  }

  total_counts

```